crypto/dh/dh_lib.c

```diff
@@ -258,8 +261,11 @@ int DH_set0_key(DH *dh, BIGNUM *pub_key, BIGNUM *priv_key)
         dh->pub_key = pub_key;
     }
     if (priv_key != NULL) {
+        int bits = BN_num_bits(priv_key);
+
         BN_clear_free(dh->priv_key);
         dh->priv_key = priv_key;
+        dh->length = bits;
```

Comment:
It strikes me as a bit odd to use a temporary variable here when you don't up in the `if (q != NULL)` body.

Reply:
Ahh yeah I changed my mind half way through - will fix..